client/src/main/java/com/netflix/conductor/client/task/WorkflowTaskCoordinator.java

```diff
@@ -275,11 +269,17 @@ private void pollForTask(Worker worker) {
 		try{
 			
 			String taskType = worker.getTaskDefName();
-			Stopwatch sw = WorkflowTaskMetrics.pollTimer(worker.getTaskDefName());
-			List<Task> tasks = client.poll(taskType, domain, worker.getIdentity(), worker.getPollCount(), worker.getLongPollTimeoutInMS());
-			sw.stop();
-			logger.debug("Polled {}, for domain {} and receivd {} tasks", worker.getTaskDefName(), domain, tasks.size());
-			for(Task task : tasks) {
+            List<Task> tasks = Lists.newArrayList();
```

Comment:
How about using logic similar to https://github.com/Netflix/conductor/blob/master/core/src/main/java/com/netflix/conductor/core/execution/tasks/SystemTaskWorkerCoordinator.java#L137 ?

Currently, the internal queue is not stored as a class variable, but it can be and then its size can be used to find out if the queue is full and if so, do not poll.